airflow/utils/dates.py

```diff
@@ -225,3 +226,14 @@ def days_ago(n, hour=0, minute=0, second=0, microsecond=0):
         second=second,
         microsecond=microsecond)
     return today - timedelta(days=n)
+
+
+def to_iso(execution_date):
+    """ Parse execution date to iso format. """
+    iso = execution_date
+    try:
+        iso = execution_date.isoformat()
+    except AttributeError:
```

Comment:
If it's not possible to call different functions depending on the type at the call sites, I think rather than assume an arbitrary type and the assume another arbitrary type we should check the type of the date and then call the right method as appropriate (or throw an exception for the unknown type). You can do this with isinstance(object, type).